core/src/main/scala/stryker4s/mutants/findmutants/FileCollector.scala

```diff
@@ -1,29 +1,80 @@
 package stryker4s.mutants.findmutants
 
 import better.files._
+import grizzled.slf4j.Logging
 import stryker4s.config.Config
+import stryker4s.run.process.{Command, ProcessRunner}
+
+import scala.util.{Failure, Success}
 
 trait SourceCollector {
   def collectFilesToMutate(): Iterable[File]
+  def filesToCopy(processRunner: ProcessRunner): Iterable[File]
 }
 
-class FileCollector(implicit config: Config) extends SourceCollector {
+class FileCollector(implicit config: Config) extends SourceCollector with Logging {
 
+  /**
+    *  Collect all files that are going to be mutated.
+    */
   override def collectFilesToMutate(): Iterable[File] = {
-    filesToMutate.filterNot(file => filesToExcludeFromMutation.contains(file))
+    filesToMutate
+      .filterNot(filesToExcludeFromMutation.contains(_))
+      .filterNot(stryker4sTmpFiles.contains(_))
   }
 
-  private[this] val filesToMutate: Seq[File] = {
-    config.mutate
-      .filterNot(file => file.startsWith("!"))
+  /**
+    * Collect all files that are needed to be copied over to the Stryker4s-tmp folder.
+    *
+    * Option 1: Copy every file that is listed by git.
+    * Option 2: Copy every file that is listed by the 'files' config setting
+    * Option 3: Copy every file in the 'baseDir' excluding target folders.
+    */
+  override def filesToCopy(processRunner: ProcessRunner): Iterable[File] = {
+    processRunner(Command("git ls-files", "--others --exclude-standard --cached"), config.baseDir) match {
+      case Success(files) => files.map(config.baseDir / _).distinct
+      case Failure(_) =>
+        info("Not a git repo falling back to 'files' configuration.")
+        fallBackToFilesConfiguration()
+    }
+  }
+
+  private[this] def fallBackToFilesConfiguration(): Seq[File] = {
+    config.files match {
+      case Some(files) => glob(files)
+      case None =>
+        warn("No 'files' specified falling back to copying everything excluding target")
```

Comment:
Should have a comma after 'specified', and changing the text to something like 'copying everything except for the target/ folder' might be clearer